Update of java action resulted in 503 failure and action remains unmanagable

## Environment details:

* Bluemix 

## Steps to reproduce the issue:

1. Delete a Java action  
2. Create a Java action

## Provide the actual results and outputs:

```
$ bx wsk action invoke actionX --result
{
    "error": "Action could not be found or may have been deleted."
}
$ bx wsk action delete actionX
error: Unable to delete action 'actionX': The connection failed, or timed out. (HTTP status code 404)
Run 'wsk --help' for usage.
```

The error shown in the controller looks as follows:
```
[2018-06-15T11:08:17.039Z] [ERROR] [#tid_xxxxxx] [CouchDbRestStore] [ATT_PUT] 'xxxxxx' internal error, name: xxxxx', doc: 'id: xxxxxx, rev: xxxxxx', failure: 'Substream Source cannot be materialized more than once'

[2018-06-15T11:06:56.634Z] [INFO] [#tid_xxxxxx] [WhiskAction] write initiated on existing cache entry, invalidating CacheKey(xxxxxx/xxxxxx), tid_xxxxxx, state WriteInProgress

[2018-06-15T11:06:56.581Z] [INFO] [#tid_xxxxxx] [WhiskAction] write all done, caching CacheKey(xxxxxx/xxxxxx) Cached

[2018-06-15T11:06:56.581Z] [INFO] [#tid_xxxxxx] [WhiskAction] write initiated on new cache entry

[2018-06-15T11:06:55.590Z] [INFO] [#tid_xxxxxx] [RestAPIVersion] authenticate: xxxxxx
```

## Additional information you deem important:
* The system runs with two controller instances
* The issue is reproducible
* The user is not able to delete nor update the action

--- a/common/scala/src/main/resources/application.conf
+++ b/common/scala/src/main/resources/application.conf
@@ -117,7 +117,7 @@ whisk {
 
         # Size limit for inlined attachments. Attachments having size less than this would
         # be inlined with there content encoded in attachmentName
-        max-inline-size = 16 k
+        max-inline-size = 0 k
 
         # Chunk sized for converting source of bytes to ByteString as part of attachment
         # upload flow

--- a/common/scala/src/main/scala/whisk/core/database/CouchDbRestStore.scala
+++ b/common/scala/src/main/scala/whisk/core/database/CouchDbRestStore.scala
@@ -372,24 +372,34 @@ class CouchDbRestStore[DocumentAbstraction <: DocumentSerializer](dbProtocol: St
     update: (A, Attached) => A,
     contentType: ContentType,
     docStream: Source[ByteString, _])(implicit transid: TransactionId) = {
-    for {
-      (bytes, tailSource) <- inlineAndTail(docStream)
-      uri <- Future.successful(uriOf(bytes, UUID().asString))
-      attached <- {
-        val a = if (isInlined(uri)) {
-          Attached(uri.toString, contentType, Some(bytes.size), Some(digest(bytes)))
-        } else {
-          Attached(uri.toString, contentType)
+
+    if (maxInlineSize.toBytes == 0) {
+      val uri = Uri.from(scheme = attachmentScheme, path = UUID().asString)
+      for {
+        attached <- Future.successful(Attached(uri.toString, contentType))
+        i1 <- put(update(doc, attached))
+        i2 <- attach(i1, uri.path.toString, attached.attachmentType, docStream)
+      } yield (i2, attached)
+    } else {
+      for {
+        (bytes, tailSource) <- inlineAndTail(docStream)
+        uri <- Future.successful(uriOf(bytes, UUID().asString))
+        attached <- {
+          val a = if (isInlined(uri)) {
+            Attached(uri.toString, contentType, Some(bytes.size), Some(digest(bytes)))
+          } else {
+            Attached(uri.toString, contentType)
+          }
+          Future.successful(a)
         }
-        Future.successful(a)
-      }
-      i1 <- put(update(doc, attached))
-      i2 <- if (isInlined(uri)) {
-        Future.successful(i1)
-      } else {
-        attach(i1, uri.path.toString, attached.attachmentType, combinedSource(bytes, tailSource))
-      }
-    } yield (i2, attached)
+        i1 <- put(update(doc, attached))
+        i2 <- if (isInlined(uri)) {
+          Future.successful(i1)
+        } else {
+          attach(i1, uri.path.toString, attached.attachmentType, combinedSource(bytes, tailSource))
+        }
+      } yield (i2, attached)
+    }
   }
 
   private def attachToExternalStore[A <: DocumentAbstraction](

--- a/tests/src/test/scala/whisk/core/controller/test/ActionsApiTests.scala
+++ b/tests/src/test/scala/whisk/core/controller/test/ActionsApiTests.scala
@@ -859,6 +859,7 @@ class ActionsApiTests extends ControllerTestCommon with WhiskActionsApi {
   }
 
   it should "put and then get an action with inlined attachment" in {
+    assumeAttachmentInliningEnabled(entityStore)
     val action =
       WhiskAction(
         namespace,

--- a/tests/src/test/scala/whisk/core/database/test/DbUtils.scala
+++ b/tests/src/test/scala/whisk/core/database/test/DbUtils.scala
@@ -24,6 +24,7 @@ import java.util.concurrent.atomic.AtomicInteger
 import akka.http.scaladsl.model.ContentType
 import akka.stream.scaladsl.Source
 import akka.util.ByteString
+import org.scalatest.Assertions
 import spray.json.DefaultJsonProtocol._
 import spray.json._
 import whisk.common.TransactionId
@@ -45,7 +46,7 @@ import scala.util.{Failure, Random, Success, Try}
  * operations with those that flow through the cache. To mitigate this, use unique asset
  * names in tests, and defer all cleanup to the end of a test suite.
  */
-trait DbUtils {
+trait DbUtils extends Assertions {
   implicit val dbOpTimeout = 15 seconds
   val instance = InstanceId(0)
   val docsToDelete = ListBuffer[(ArtifactStore[_], DocInfo)]()
@@ -314,6 +315,10 @@ trait DbUtils {
       case _ =>
         42
     }
+  }
+
+  def assumeAttachmentInliningEnabled(db: ArtifactStore[_]): Unit = {
+    assume(inlinedAttachmentSize(db) > 0, "Attachment inlining is disabled")
   }
 
   protected def encodedRandomBytes(size: Int): String = Base64.getEncoder.encodeToString(randomBytes(size))

--- a/tests/src/test/scala/whisk/core/database/test/behavior/ArtifactStoreAttachmentBehaviors.scala
+++ b/tests/src/test/scala/whisk/core/database/test/behavior/ArtifactStoreAttachmentBehaviors.scala
@@ -90,6 +90,7 @@ trait ArtifactStoreAttachmentBehaviors extends ArtifactStoreBehaviorBase with Ex
    * if attachment is inlined
    */
   it should "work on reading with old inlined attachment" in {
+    assumeAttachmentInliningEnabled(entityStore)
     implicit val tid: TransactionId = transid()
     val code1 = encodedRandomBytes(inlinedAttachmentSize(entityStore))
     val exec = javaDefault(code1, Some("hello"))
@@ -140,6 +141,7 @@ trait ArtifactStoreAttachmentBehaviors extends ArtifactStoreBehaviorBase with Ex
   }
 
   it should "inline small attachments" in {
+    assumeAttachmentInliningEnabled(entityStore)
     implicit val tid: TransactionId = transid()
     val attachmentSize = inlinedAttachmentSize(entityStore) - 1
     val base64 = encodedRandomBytes(attachmentSize)

